terraform/bigquery.tf

```diff
@@ -740,3 +740,90 @@ resource "google_bigquery_table_iam_member" "commit_review_status_viewers" {
   role       = "roles/bigquery.dataViewer"
   member     = each.value
 }
+
+# Invocation Comment Status Table / IAM
+
+resource "google_bigquery_table" "invocation_comment_table" {
```

Comment:
How does this table work? Are you writing a new row for each invocation attempt?

If so, then what is the purpose of the job_attempts column?

If your intention is to update this row with the latest job run then I would suggest rethinking that. BQ does not love updates.

Reply:
job_attempts is for if you fail and need to retry. We don't want to retry forever, so after it is incremented a few times we will no longer attempt.

Reply:
So each run will lookup the previous run's value and increment it for this run? Why not use a select count by PR id?

Reply:
That was my thought, but I guess select count by PR id would work too. It makes the selection query messier though.

Reply:
What criteria would you use to look up the last row? It would be similar, right? Select max(job_attempts) ...

Not sure one is better than the other but a count would eliminate the need to store the value.

Reply:
So with the schema as it currently is, I can do something like

```sql
SELECT pull_request_id
FROM invocation_comment status
WHERE status = "SUCCESS" OR  job_attempts >= 3
```

to get the list of PRs I do not want to process, because they already succeeded or have failed too many times.

You could do the same with multiple rows, but the query would be more complex. You would need to group by pull_request_id, then have a having clause to limit to PRs with less than two rows, but I don't know how you could filter ones with success without a nested query of some sort.

Reply:
https://cloud.google.com/bigquery/docs/reference/standard-sql/dml-syntax

Though I do see what you mean about updates being costly. `invocation_comment_status` should be a pretty small table (maybe like 10 MB) but maybe that isn't ideal.

Reply:
I think I figured out a good query to do what I want
```sql
  HAVING COUNT(*) >= 3 OR COUNTIF(ics.status = "SUCCESS") > 0
  ```

Reply:
Let's go ahead and remove this field, I think we can do without